src/faldbt/parse.py

```diff
@@ -174,22 +174,27 @@ def get_global_script_configs(source_dirs: List[Path]) -> Dict[str, List[str]]:
     return global_scripts
 
 
-# TODO: temporary until the fal runtime work gets merged in
-def get_environment(base_dir: str, name: str) -> "Environment":
+def load_environments(base_dir: str) -> Dict[str, "BaseEnvironment"]:
     from fal.packages.environment import create_environment
+    from fal.fal_script import _is_local_environment
 
     fal_project_path = os.path.join(base_dir, "fal_project.yml")
     if not os.path.exists(fal_project_path):
         raise FalParseError("{fal_project_path} must exist to define environments")
 
     fal_project = load_yaml(fal_project_path)
+
+    environments = {}
     for environment in fal_project.get("environments", []):
-        if environment["name"] == name:
-            return create_environment(requirements=environment.get("requirements", []))
-    else:
-        raise FalParseError(
-            f"Could not find the environment definition for {name} in {fal_project_path}"
-        )
+        if "name" not in environment:
+            raise FalParseError("Environment must have a name")
+
+        name = environment.get("name")
+        if _is_local_environment(name):
+            raise FalParseError(f"Environment name conflicts with a reserved name: {name}.")
```

Comment:
How do we feel of making the `local` environment actually a `null`?

https://www.educative.io/answers/how-to-represent-null-values-in-yaml

```yaml
        post-hook:
          - path: environment_type_2.py
            environment: null
```


```yaml
        post-hook:
          - path: environment_type_2.py
            environment: ~
```


```yaml
        post-hook:
          - path: environment_type_2.py
            environment:
```

Reply:
I do like how `local` is explicit. But I don't know, just a thought.

Reply:
@mederka @turbo1912 @burkaygur any preferences? It is kind of annoying that this prevents people from naming their own environments `local` but I feel just like @chamini2 in the sense that it is more explicit than `~`.

Reply:
Let's merge this as is, and if anyone has any extra thoughts on this we can address them separately on https://linear.app/features-and-labels/issue/FEA-372/decide-on-local-vs-for-local-environments.